src/main/java/graphql/execution/ExecutorServiceExecutionStrategy.java

```diff
@@ -58,8 +72,9 @@ public ExecutorServiceExecutionStrategy(ExecutorService executorService, DataFet
             ExecutionStrategyParameters newParameters = parameters
                     .transform(builder -> builder.field(currentField).path(fieldPath));
 
-            Callable<ExecutionResult> resolveField = () -> resolveField(executionContext, newParameters).join();
-            futures.put(fieldName, executorService.submit(resolveField));
+            Callable<ExecutionResult> fieldResolver = () -> resolveField(executionContext, newParameters).join();
+            Callable<ExecutionResult> wrappedFieldResolver = callableWrapper.wrapCallable(fieldResolver);
+            futures.put(fieldName, executorService.submit(wrappedFieldResolver));
         }
```

Comment:
That would certainly work with the current implementation of `ExecutorServiceExecutionStrategy`. The downside is that there is no explicit contract in `ExecutorServiceExecutionStrategy` to only use `ExecutorService#submit(Callable)`. If one of the other `#submit(Runnable[...])` methods were to be used, this proposed strategy would be silently broken.

Essentially, this strategy relies on end users having up-to-date knowledge of the internals of the `ExecutorServiceExecutionStrategy`, which isn't ideal.

```
    public class WrappingThreadPoolExecutor extends ThreadPoolExecutor {

        private CallableWrapper wrapper;

        public WrappingThreadPoolExecutor(CallableWrapper wrapper) {
            super(
                    0,
                    100,
                    60L, TimeUnit.SECONDS,
                    new SynchronousQueue<>(true)
            );

            this.wrapper = wrapper;
        }

        @Override
        public <T> Future<T> submit(Callable<T> task) {
            return super.submit(wrapper.wrapCallable(task));
        }
    }
```

If this strategy is to be used, it would be better if `ExecutorServiceExecutionStrategy` were injected with some `CallableOnlyExecutorService` which wraps an `ExecutorService` to provide that contract.